### DISCUSSION TITLE
xDrip+ Crashes to Desktop on Galaxy Watch5 Pro

### DISCUSSION THREAD

**Discussion by WiTuLo** [NightscoutFoundation/xDrip#3387] [Bug reports] [answered]
For future reference: Use wear os installer (ensuring the watch keeps the wifi active during the install process). Solved.

Hi, 
When I open XDrip+ on my Galaxy Watch 5Pro after accepting the 2 disclaimers, similar to the following description:
https://github.com/NightscoutFoundation/xDrip/discussions/2969
When I try to force it open the screen turns black for a second. I didn't find any resolution (besides the self-compilation in the link above, which I'm uncomfortable with due to never having compiled anything in android studio). Since this is a relatively popular and convenient watch... did anyone find a resolution / easier workaround to this issue?

Initial attempts to bug-fix: 
- restarted phone / watch
- re-installed (split-apk and full apk via GeminiMan software)
- installed latest nightly. 
- toggled various wear options on phone Xdrip+ app
-> No change in software behaviour.

Verified on Nightly 2023114 + 20240306 (nightly build on phone corresponded w/ watch build)
Phone: Fairphone 5
Watch: Galaxy Watch5 Pro
Transfer Software: GeminiMan WearOS Manager (edit: necessary bc Wear Installer 2 refuses transfer)

Any ideas appreciated, happy to try things and report back :)

**Comment by PhilTU** [comment]
Just a passive observation - people seem to succeed using Wear Installer 2 to install the watch WearOS compeonent apk so may be worth a try.

**Reply by WiTuLo** [reply]
Sorry should have mentioned - Wear installer 2 doesn't transfer files for my setup. Editing post to clarify.

**Reply by PhilTU** [reply, marked as answer]
interesting, when you say it "refuses transfer" this seems at odds with the experience of others. Perhaps the other tool is having issues as well. Unfortunately the WearOS support is a bit limited currently.

**Reply by WiTuLo** [reply]
This actually turned out to be the solution... I managed to re-install on the watch with wear installer 2 today. Apparently something repeatedly failed to install during the GeminiMan install process. I think that the watch needed to be forced to keep the wireless connection up for this to work (I did this by repeatedly tapping the "connected device" during the transfer phase). Awfully sorry for wasting yours and @Navid200 's time!!!

**Reply by PhilTU** [reply]
Excellent, glad that worked. It wasn't a waste of time.

There's something I haven't fully got my head round yet where pairing for wireless adb via wifi is different bluetooth and so on since an update to WearOS.

**Comment by Navid200** [comment]
What is the meaning of desktop when talking about a watch?

**Reply by WiTuLo** [reply]
The watch goes back to the app selection screen. The watch still thinks that it is running: Its symbol is shown in the "running apps" display. However, the display only shows the xdrip-logo. Same for the complication in the actual watchface: A logo is shown permanently (even when choosing "close all" on the app selection screen), but now data is ever provided. Toggeling the "reset wear DB" or "open wear setttings on watch" items on the phone also causes no reaction on the watch (the phne app also doesn't launch because of any actions on the phone, don't know if this is normal).

**Reply by PhilTU** [reply]
So you can't access xDrip+ menu items on the watch ? https://xdrip.readthedocs.io/en/latest/smartwatch/usewear/#discover-wear-extension
